frontend/client/src/screens/Login.js

```diff
@@ -58,9 +58,9 @@ const SignInFormBase = observer(
       // Based on https://firebase.google.com/docs/auth/web/email-link-auth
       if (this.props.store.isSignInWithEmailLink(window.location.href)) {
         Logging.log('signInWithEmailLink detected')
+        var email = new URL(location.href).searchParams.get('email')
         // Get the email if available. This should be available if the user completes
         // the flow on the same device where they started it.
-        var email = window.localStorage.getItem('emailForSignIn')
         if (!email) {
           // User opened the link on a different device. To prevent session fixation
           // attacks, ask the user to provide the associated email again. For example:
```

Comment:
This doesn't work anymore. Now that we are reading the email from the URL instead of local storage, we cannot guarantee that the user is logging in from the device that generated the magic link.